sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java

```diff
@@ -259,43 +262,89 @@ public void cancel(Context context, long transferId) {
     /**
      * Creates a new container within a storage account. If a container with the same name already exists, the operation
      * fails.
+     * For more information, see the <a href="https://docs.microsoft.com/rest/api/storageservices/create-container">Azure Docs</a>.
      *
      * @param containerName The container name.
      * @param callback      Callback that receives the response.
      */
-    public void createContainer(String containerName,
-                                CallbackWithHeader<Void, ContainerCreateHeaders> callback) {
-        storageBlobServiceClient.createContainer(containerName, callback);
+    public void createContainer(@NonNull String containerName,
+                                @Nullable CallbackWithHeader<Void, ContainerCreateHeaders> callback) {
+        this.createContainer(new ContainerCreateOptions(containerName), callback);
     }
 
     /**
      * Creates a new container within a storage account. If a container with the same name already exists, the operation
      * fails.
+     * For more information, see the <a href="https://docs.microsoft.com/rest/api/storageservices/create-container">Azure Docs</a>.
      *
-     * @param containerName     The container name.
-     * @param timeout           The timeout parameter is expressed in seconds. For more information, see
-     *                          &lt;a href="https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations"&gt;Setting Timeouts for Blob Service Operations.&lt;/a&gt;.
-     * @param metadata          Metadata to associate with the container.
-     * @param publicAccessType  Specifies how the data in this container is available to the public. See the
-     *                          x-ms-blob-public-access header in the Azure Docs for more information. Pass null
-     *                          for no public access.
-     * @param cancellationToken The token to request cancellation.
-     * @param callback          Callback that receives the response.
-     * @return The response information returned from the server when creating a container.
+     * @param options {@link ContainerCreateOptions}
+     * @param callback Callback that receives the response.
      */
-    public void createContainer(String containerName,
-                                Integer timeout,
-                                Map<String, String> metadata,
-                                PublicAccessType publicAccessType,
-                                CancellationToken cancellationToken,
-                                CallbackWithHeader<Void, ContainerCreateHeaders> callback) {
+    public void createContainer(@NonNull ContainerCreateOptions options,
+                                @Nullable CallbackWithHeader<Void, ContainerCreateHeaders> callback) {
+        Objects.requireNonNull(options);
+        storageBlobServiceClient.createContainer(options.getContainerName(), options.getTimeout(),
```

Comment:
Why expand the options here rather than in the Intern method?

Reply:
It's future proofing for when we generate code. The intern methods will eventually be generated so we probably want to do all this at the client level